online-shop/src/app/app.module.ts

```diff
@@ -11,6 +11,12 @@ import { IconButtonComponent } from './modules/shared/components/presentational/
 import { ShoppingCartDetailsViewComponent } from './modules/shopping-cart/components/presentational/shopping-cart-details-view/shopping-cart-details-view.component';
 import { ShoppingCartDetailsComponent } from './modules/shopping-cart/components/containers/shopping-cart-details/shopping-cart-details.component';
 import { ShoppingCartModule } from './modules/shopping-cart/shopping-cart.module';
+import { ProductsFormComponent } from './components/containers/products-form/products-form.component';
+import { ProductsEditFormComponent } from './components/containers/products-edit-form/products-edit-form.component';
+import { ProductsFormViewComponent } from './components/presentational/products-form-view/products-form-view.component';
+import { ProductsEditFormViewComponent } from './components/presentational/products-edit-form-view/products-edit-form-view.component';
```

Comment:
Do you really need a form for each action (add & edit)? You can basically use just one single component for it, and based on the route path (add/... edit/...) you can decide the logic inside it.

Please check the following folder structure:
```
├── src/app
│   ├── components
│   │   ├── containers
│   │   │   ├── products-form
│   │   │   │   ├── products-form.component.ts
│   │   │   │   ├── products-form.component.scss
│   │   │   │   ├── products-form.component.html
│   │   ├── presentational
│   │       ├── products-form-view
│   │       │   ├── products-form-view.component.ts
│   │       │   ├── products-form-view.component.scss
│   │       │   ├── products-form-view.component.html
│.......<the previous files>................................
```